pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -760,6 +792,36 @@ func validateValidation(v *admissionregistration.Validation, paramKind *admissio
 	return allErrors
 }
 
+func validateAuditAnnotation(meta metav1.ObjectMeta, v *admissionregistration.AuditAnnotation, paramKind *admissionregistration.ParamKind, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if len(meta.GetName()) != 0 {
+		name := meta.GetName()
+		allErrors = append(allErrors, apivalidation.ValidateQualifiedName(name+"/"+v.Key, fldPath.Child("key"))...)
+	} else {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("key"), v.Key, "requires metadata.name be non-empty"))
+	}
+
+	trimmedValueExpression := strings.TrimSpace(v.ValueExpression)
```

Comment:
I was following what we do already for `Validations.Expression`: https://github.com/kubernetes/kubernetes/blob/b86f94f438b7e62a8dc6446cbdfa224bb4465f37/pkg/apis/admissionregistration/validation/validation.go#L731

This seems sane for CEL expressions since an expression will fail to parse if empty.

Reply:
we trimmed the main CEL expression for two reasons:
1. because it was going to be used as the default error message in failure cases, and we wanted to require a separate human-readable message if the CEL expression contained newlines after trimming
2. making sure it was non-empty after trimming space was a cheap check before getting into CEL parsing

1 doesn't seem to apply here. 2 does.